src/common/date-provider.ts

```diff
@@ -2,7 +2,7 @@
 // Licensed under the MIT License.
 export class DateProvider {
     public static getDate(timestamp?: string): Date {
-        return new Date(timestamp);
+        return timestamp == null ? new Date() : new Date(timestamp);
```

Comment:
I guess is time to add some sort of test around this? is not longer a wrapper around `new Date`